ui-v2/src/lib/types.ts

```diff
@@ -5,3 +5,8 @@ export type JSONValue =
 	| Record<string, never>
 	| unknown[]
 	| null;
+
+export type ReactComponentPropsWithClassName<T extends React.ElementType> =
```

Comment:
This was added to make edits in the [alert-dialog](https://github.com/j-brenneman/prefect-fork/pull/1/files#diff-796cae0c3dd848d9c4a49acbdaf20acc328eb25e8b9bba6ad70356b24efcb064) from shadcn easier.

When copy/pasting there are initially typescript errors, because the composable components of the AlertDialog need a className prop in addition to the radix primitive props. 

Seems this will be a common problem whenever importing shadcn components, and this custom type allows a one line change on each component of `ReactComponentPropsWithClassName<typeof AlertDialogPrimitive.Action>` instead of having to define explicit prop types for each above the component.